PyInstaller/compat.py

```diff
@@ -870,10 +870,13 @@ def check_requirements():
         if 'win32api' in sys.modules or 'pywintypes' in sys.modules:
             # Users should never see this error; if it occurs, it means someone
```

Comment:
I cannot understand why this approach would succeed when the other approach would fail. Did you try simply putting `check_requirements()` at the end of the file?